src/services/crypto.service.ts

```diff
@@ -580,8 +591,35 @@ export class CryptoService implements CryptoServiceAbstraction {
         return min + rval;
     }
 
+    async validateKey(key: SymmetricCryptoKey) {
+        try {
+            const encPrivateKey = await this.storageService.get<string>(Keys.encPrivateKey);
+            if (encPrivateKey == null) {
+                return false;
+            }
+
+            const encKey = await this.getEncKey(key);
+            const privateKey = await this.decryptToBytes(new EncString(encPrivateKey), encKey);
+            await this.cryptoFunctionService.rsaExtractPublicKey(privateKey);
+        } catch (e) {
+            return false;
+        }
+
+        return true;
+    }
+
     // Helpers
 
+    private async shouldStoreKey() {
+        const vaultTimeout = await this.storageService.get<number>(ConstantsService.vaultTimeoutKey);
+        const biometricUnlock = await this.storageService.get<boolean>(ConstantsService.biometricUnlockKey);
+
+        const biometricsEnabled = biometricUnlock && this.platformUtilService.supportsSecureStorage();
+        const noVaultTimeout = vaultTimeout == null;
+
+        return noVaultTimeout || biometricsEnabled;
```

Comment:
This hurt my brain, but boolean logic refactor is correct from what I can see as I walk through it.